Common/Mechanics/SkillTags.cs

```diff
@@ -0,0 +1,27 @@
+﻿namespace PathOfTerraria.Common.Mechanics;
+
+[Flags]
+public enum SkillTags : ulong
+{
+	None = 0,
+	Fire = 1,
+	Cold = 2,
+	Lightning = 4,
+	Chaos = 8,
+	Melee = 16,
+	Ranged = 32,
+	Magic = 64,
+	Summon = 128,
+	Projectile = 256,
+	Channeling = 512,
+	AreaOfEffect = 1024,
+	Sentry = 2048,
+	Minion = 4096,
+	Debuff = 8192,
+	Buff = 16384,
+	Movement = 32768,
+	Critical = 65536,
+	Healing = 131072,
+
+	Elemental = Fire | Cold | Lightning | Chaos,
+}
```

Comment:
```suggestion
[Flags]
public enum SkillTags : ulong
{
	None            = 0,
	Fire            = 1 << 0,
	Cold            = 1 << 1,
	Lightning       = 1 << 2,
	Chaos           = 1 << 3,
	Melee           = 1 << 4,
	Ranged          = 1 << 5,
	Magic           = 1 << 6,
	Summon          = 1 << 7,
	Projectile      = 1 << 8,
	Channeling      = 1 << 9,
	AreaOfEffect    = 1 << 10,
	Sentry          = 1 << 11,
	Minion          = 1 << 12,
	Debuff          = 1 << 13,
	Buff            = 1 << 14,
	Movement        = 1 << 15,
	Critical        = 1 << 16,
	Healing         = 1 << 17,

	Elemental       = Fire | Cold | Lightning,
}
```

Couple things.
1) Bit shifting much easier to read/manage
2) Chaos damage is actually not considered elemental, it's own beast.
For example if "+5% All Elemental Resistances" - You would not get chaos resistance.